galaxyui/src/app/my-content/namespace-detail/namespace-detail.component.ts

```diff
@@ -0,0 +1,127 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { Router, ActivatedRoute } from '@angular/router';
+
+import { cloneDeep } from 'lodash';
+
+import {
+    FormBuilder,
+    FormControl,
+    FormGroup,
+    Validators
+} from '@angular/forms';
+
+import { Namespace }         from '../../resources/namespaces/namespace';
+
+import { NamespaceService }      from "../../resources/namespaces/namespace.service";
+import { UserService }           from "../../resources/users/user.service";
+import { User }                  from "../../resources/users/user";
+import { ProviderNamespace }     from "../../resources/provider-namespaces/provider-namespace";
+import { ProviderSourceService } from "../../resources/provider-namespaces/provider-source.service";
+
+
+@Component({
+    selector: 'app-namespace-detail',
+    templateUrl: './namespace-detail.component.html',
+    styleUrls: ['./namespace-detail.component.less']
+})
+export class NamespaceDetailComponent implements OnInit {
+    @Input() namespace: Namespace;
+
+    namespaceForm: FormGroup;
+    // namespace: Namespace = new Namespace();
+    users: User[];
+    providerSources: ProviderNamespace[];
+    id: number;
+
+    constructor(private route: ActivatedRoute,
+                private router: Router,
+                private formBuilder: FormBuilder,
+                private namespaceService: NamespaceService,
+                private userService: UserService,
+                private providerSourceService: ProviderSourceService) {
+    }
+
+    ngOnInit() {
+        this.route.data
+            .subscribe((data: { namespace: Namespace }) => {
+                this.namespace = data.namespace ? data.namespace : new Namespace();
+                console.log('namespace detail ngOnInit', this.namespace);
+                this.createForm();
```

Comment:
TODO: probably should have done a createForm in the constructor and then done a reset form in ngInit.